pkgs/games/factorio/fetch.sh

```diff
@@ -14,42 +14,32 @@ curl="curl \
  $curlOpts \
  $NIX_CURL_FLAGS"
 
-# We don't want the password to be on any program's argv, as it may be
+# We don't want the token to be on any program's argv, as it may be
 # visible in /proc. Writing it to file with echo should be safe, since
 # it's a shell builtin.
-echo -n "$password" > password
+echo -n "$token" > token
 # Might as well hide the username as well.
 echo -n "$username" > username
 
-# Get a CSRF token.
-csrf=$($curl $loginUrl | xidel - -e '//input[@id="csrf_token"]/@value')
+$curl --get \
+--data-urlencode username@username \
+--data-urlencode token@token \
+--location --fail $url > $out || {
+  set +x
+  echo <<EOF
+Fetch failed.
+Please ensure you have set the username and token with config.nix,'
+or /etc/nix/nixpkgs-config.nix if on NixOS.'
 
-# Log in. We don't especially care about the result, but let's check if login failed.
-$curl --data-urlencode csrf_token="$csrf" \
-      --data-urlencode username_or_email@username \
-      --data-urlencode password@password \
-      -d action=Login \
-      $loginUrl -D headers > /dev/null
-
-if grep -q 'Location: https://' headers; then
-    # Now download. We need --insecure for this, but the sha256 should cover us.
-    $curl --insecure --location --fail $url > $out || { echo "Login succeeded, but subsequent fetch failed."; exit 1; }
-    set +x
-else
-    set +x
-    echo 'Login failed'
-    echo 'Please set username and password with config.nix,'
-    echo 'or /etc/nix/nixpkgs-config.nix if on NixOS.'
-    echo
-    echo 'Example:'
-    echo '{'
-    echo '  packageOverrides = pkgs: rec {'
-    echo '    factorio = pkgs.factorio.override {'
-    echo '      username = "<username or email address>";'
-    echo '      password = "<password>";'
-    echo '    };'
-    echo '  };'
-    echo '}'
-
-    exit 1
-fi
+Example:
+{
+  packageOverrides = pkgs: rec {
+    factorio = pkgs.factorio.override {
+      username = "<username or email address>";
```

Comment:
nit: Insert a comment saying that this is confidential data and that it should not be shared.
follow-up: Use overlays instead of packageOverrides.